### PR TITLE
[Merged by Bors] - feat: filters that are disjoint from the cocompact filter

### PR DESCRIPTION
**Pull request by trivial1711**
- `Filter.disjoint_cocompact_left`, `Filter.disjoint_cocompact_right` : A filter is disjoint from the cocompact filter if and only if it contains a compact set.
- `nhdsSet_prod_le_of_disjoint_cocompact`, `prod_nhdsSet_le_of_disjoint_cocompact` : If `s` is a compact set and `f` is disjoint from the cocompact filter, then `𝓝ˢ s ×ˢ f ≤ 𝓝ˢ (s ×ˢ Set.univ)`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Compactness/Compact.lean [hunk 1]
```diff
@@ -612,6 +612,18 @@ theorem cocompact_eq_cofinite (X : Type*) [TopologicalSpace X] [DiscreteTopology
   simp only [cocompact, hasBasis_cofinite.eq_biInf, isCompact_iff_finite]
 #align filter.cocompact_eq_cofinite Filter.cocompact_eq_cofinite
 
+/-- A filter is disjoint from the cocompact filter if and only if it contains a compact set. -/
+theorem disjoint_cocompact_left (f : Filter X) :
```
Comment (riccardobrasca) [inline Mathlib/Topology/Compactness/Compact.lean:616]:
```suggestion
theorem cocompact.disjoint_iff_left (f : Filter X) :
```
I am not 100% sure, but I am following the naming of [Filter.HasBasis.disjoint_iff_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Bases.html#Filter.HasBasis.disjoint_iff_left).

(Similarly below).

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Topology/Compactness/Compact.lean (hunk 1 above)
Comment (trivial1711) [reply Mathlib/Topology/Compactness/Compact.lean:616]: I'm following the naming of [Filter.disjoint_cofinite_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Cofinite.html#Filter.disjoint_cofinite_left).

**Pull request closed without merging**